server/routes/api/auth.js

```diff
@@ -22,4 +25,52 @@ router.post('/login', (req, res) => {
     });
 });
 
+router.get('/logout', (req, res) => {
+  if (req.session.userId) {
+    req.session.destroy(err => {
+      err ? res.status(500).json(err) : res.status(200)
+    })
+  }
+  res.end()
+});
+
+router.post('/forgot', (req, res) => {
+  User.findOne(req.body)
+    .exec()
+    .then(user => {
+      if (!user) {
+        return res.status(200).end()
+      } else {
+        const token = crypto.randomBytes(20).toString('hex');
+        user.resetPassword.token = token
+        user.resetPassword.expiration = Date.now() + 3600000;
+        user.save()
```

Comment:
It looks like an error with this save will actually get caught at the final catch and serve up a 500. I'm not sure if that's what we want to do. This is a good opportunity for async await instead of these nested promises.

It looks like we have 3 asynchronous calls:
1) Find the user
2) Save the token onto the user
3) Sending the email

Can you think of a way to bring async await into this and better control the errors/status codes that we return?